### DISCUSSION TITLE
Is it possible to define JavaScript functions that callback to Ruby?

### DISCUSSION THREAD

**Discussion by postmodern** [rubycdp/ferrum#280] [Q&A]
According to the examples one can inject their own JavaScript into the page. Is it also possible to define a JavaScript function that calls back into Ferrum? I know this is possible with [therubyracer](https://github.com/rubyjs/therubyracer#readme) which is built on a headless V8.

**Comment by route** [comment]
If you explain the scenario I can try to answer, but now I don't understand the question honestly.

**Reply by postmodern** [reply]
With [therubyracer](https://github.com/rubyjs/therubyracer#synopsis) it is possible to define JavaScript functions as Ruby lambdas or assign Ruby objects to JavaScript objects. This allows the JavaScript to call these functions which in turn executes the Ruby lambda or method defined on the Ruby object, allowing JavaScript data to be passed back into Ruby. This would be useful for registering JavaScript event listeners or hooking JavaScript functions. Is this possible with Ferrum's API?

**Reply by route** [reply]
You can pass arguments to JS world from ruby with https://github.com/rubycdp/ferrum#evaluate_asyncexpression-wait_time-args but there's no such functionality you described
